### PR TITLE
Issue #12716: Make config filenames kebab-case

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #12716

### CODE REVIEW

File: config/checkstyle-non-main-files-suppressions.xml
Comment (stoyanK7) [inline config/checkstyle-non-main-files-suppressions.xml:174]: This file was dubious. Not sure if I should rename it and what I should rename it to.

`starter-rule-set-all-rules-by-category.groovy.txt`?

**Review by romani — APPROVED** [review]
Ok to merge if CI pass

File: config/checkstyle-non-main-files-suppressions.xml
Comment (rnveach) [inline config/checkstyle-non-main-files-suppressions.xml:174]: Missing a reason for suppression.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: config/checkstyle-non-main-files-suppressions.xml
Comment (stoyanK7) [reply config/checkstyle-non-main-files-suppressions.xml:174]: It is what I asked above. Should we suppress it at all?

Comment (romani) [reply config/checkstyle-non-main-files-suppressions.xml:174]: I am ok to change name of this file

Comment (rnveach) [reply config/checkstyle-non-main-files-suppressions.xml:174]: I do not see why it can't be renamed.
https://github.com/checkstyle/checkstyle/blob/master/.ci/codenarc.groovy#L10
It is a file of our creation we pass in by specific name.

@nrmancuso @Vyom-Yadav Any complaints against renaming this file to follow the new rules?

Comment (nrmancuso) [reply config/checkstyle-non-main-files-suppressions.xml:174]: This name is bad, please rename it. If extension doesn’t matter, I propose codenarc-rules.groovy, since I think this file is actually in groovy syntax if I remember correctly.

Comment (romani) [reply config/checkstyle-non-main-files-suppressions.xml:174]: I ok with proposed name. Existing name is name of file that we copied from from originally

Comment (stoyanK7) [reply config/checkstyle-non-main-files-suppressions.xml:174]: @nrmancuso Just `codenarc-rules.groovy`, without the `.txt` at the end? Is there a reason for this `.txt` by the way?

Comment (Vyom-Yadav) [reply config/checkstyle-non-main-files-suppressions.xml:174]: > Just codenarc-rules.groovy, without the .txt at the end? 

Just `codenarc-rules.groovy` will work, but `codenarc-rules.groovy.txt` is better as IntelliJ won't run inspections on `.groovy.txt`. On `.groovy`, the complete file is red (full of errors). 

> Is there a reason for this .txt by the way?

Not really, it is stored as `.groovy.txt` in official codenarc repo, but using just `.groovy` will also work fine.

File: .ci/validation.sh
Comment (stoyanK7) [inline .ci/validation.sh:696]: Why is it not finding the file? Local run fails too..

https://checkstyle.semaphoreci.com/jobs/7fc71ca3-4a6b-4cac-afd7-11ca7d5637b3#L786
```
cannot initialize module SuppressionFilter - Unable to find: config/checkstyle-non-main-files-suppressions.xml
```

Comment (rnveach) [reply .ci/validation.sh:696]: I am not seeing why, but this is a maven run of `sevntu-checks`. It is not a run of Checkstyle. So it doesn't make sense why we are giving sevntu files from Checkstyle.

Comment (stoyanK7) [reply .ci/validation.sh:696]: https://github.com/stoyanK7/checkstyle/blob/cbffbbaca86cf0fc8e85c192fd3feecaebd67971/.ci/validation.sh#L685-L699

Comment (rnveach) [reply .ci/validation.sh:696]: It seems odd, the error message doesn't say ../../, so I am thinking something is missing in the transfer.

Comment (stoyanK7) [reply .ci/validation.sh:696]: It does say it in a different place in the error log though
https://checkstyle.semaphoreci.com/jobs/7fc71ca3-4a6b-4cac-afd7-11ca7d5637b3#L735
```
Caused by: org.apache.tools.ant.BuildException: Unable to create Root Module: config {../../../config/checkstyle-non-main-files-checks.xml}.

```

Comment (rnveach) [reply .ci/validation.sh:696]: I see the issue now. We are passing sevntu our XML configurations so it can run them. However, our config still references our suppression file naming and not sevntu's file naming. Sevntu still has the old name, and our configuration has the new name. Hence it is not finding the file and failing.

@stoyanK7 You must apply similar changes to sevntu just like in other PRs.

Comment (stoyanK7) [reply .ci/validation.sh:696]: I suppose we are talking about those lines the `pom.xml`?
```
    <checkstyle.configLocation>https://raw.githubusercontent.com/checkstyle/checkstyle/checkstyle-${checkstyle.version}/config/checkstyle_checks.xml</checkstyle.configLocation>
    <checkstyle.nonMain.configLocation>https://raw.githubusercontent.com/checkstyle/checkstyle/checkstyle-${checkstyle.version}/config/checkstyle_non_main_files_checks.xml</checkstyle.nonMain.configLocation>
    <checkstyle.header>https://raw.githubusercontent.com/checkstyle/checkstyle/master/config/java.header</checkstyle.header>
    <checkstyle.regexp.header>https://raw.githubusercontent.com/checkstyle/checkstyle/master/config/java_regexp.header</checkstyle.regexp.header>
```

Comment (stoyanK7) [reply .ci/validation.sh:696]: Got it.

Comment (romani) [reply .ci/validation.sh:696]: we have hardcoded file:
https://github.com/checkstyle/checkstyle/blob/a1731ba2da59d713630e06309697df326cdf77d4/config/checkstyle_non_main_files_checks.xml#L12-L15

we need it to be like:
https://github.com/checkstyle/checkstyle/blob/a1731ba2da59d713630e06309697df326cdf77d4/config/checkstyle_checks.xml#L36-L38

So in this case sevntu can stay on names of file it needs to be.

File: config/checkstyle-non-main-files-suppressions.xml
Comment (stoyanK7) [reply config/checkstyle-non-main-files-suppressions.xml]: Done. Renamed.

File: .ci/validation.sh
Comment (stoyanK7) [reply .ci/validation.sh]: @rnveach We have to merge https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/989 first, before merging this:
https://checkstyle.semaphoreci.com/jobs/2c087901-7776-40ab-9a1a-30350c48a94d#L639
```
[ERROR] [checkstyle] [ERROR] /home/semaphore/checkstyle/.ci-temp/sevntu.checkstyle/sevntu-checks/config/CheckstyleFormatterForEclipse.xml:1: Filename should be in kebab-case. [kebabCaseFileNamingInConfigFolder]
[ERROR] [checkstyle] [ERROR] /home/semaphore/checkstyle/.ci-temp/sevntu.checkstyle/sevntu-checks/config/checkstyle_non_main_files_suppressions.xml:1: Filename should be in kebab-case. [kebabCaseFileNamingInConfigFolder]
```

**Comment by stoyanK7** [comment]
@rnveach Ping. This one is ready. `No Error Test (openjdk11, fast pool) - CMD=.ci/validation.sh no-error-sevntu-checks` will fix when this gets merged.
https://checkstyle.semaphoreci.com/jobs/3d4d5cf8-bfcc-43cb-9a75-7445f9d48660#L732
`Unable to create Root Module: config {../../../config/checkstyle-checks.xml}. `
`unable to load header file https://raw.githubusercontent.com/checkstyle/checkstyle/master/config/java-regexp.header`

**Comment by romani** [comment]
need to make CI green sooner, if any items to improve, please share , we can improve in next PR.

**Pull request merged**
